src/service/test_case.rs

```diff
@@ -0,0 +1,38 @@
+use crate::entities::{Scope, TestCase};
+use crate::{CsesApi, Filesystem, Resources, Storage, RP};
+use anyhow::{Context, Result};
+
+pub fn get_test_cases(res: &mut Resources<impl RP>, scope: &Scope, task_id: &str) -> Result<()> {
+    let test_cases = fetch_test_cases(res, scope, task_id)?;
+    save_test_cases(res, test_cases)?;
+    Ok(())
+}
```

Comment:
Should this be part of the `ui` and `fetch_test_cases` and `save_test_cases` made public?

 I think it would be nice if the `ui` could ask the user for a confirmation in case some existing files are going be overwritten in which case `ui` would need to print in between `fetch_test_cases` and `save_test_cases` calls.